packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts

```diff
@@ -76,7 +76,11 @@ async function getRulesets(
 }> {
   const rulesetPayload: SpectralRulesetPayload = await getRulesetPayload(initiator, resolvedOpenapiType)
   const namesOfRulesInStagingOnly: string[] = getNamesOfRulesInPayloadWithPropertySetToTrue(rulesetPayload, "stagingOnly")
-  const namesOfRulesDisabledForTypespec: string[] = getNamesOfRulesInPayloadWithPropertySetToTrue(rulesetPayload, "disableForTypeSpec")
+  const namesOfRulesDisabledForTypespec: string[] =
+    // We ignore "disableForTypeSpec" for ARM and RPaaS rulesets.
+    resolvedOpenapiType === OpenApiTypes.arm || resolvedOpenapiType === OpenApiTypes.rpaas
+      ? []
+      : getNamesOfRulesInPayloadWithPropertySetToTrue(rulesetPayload, "disableForTypeSpec")
```

Comment:
Let's reverse the check (opt-in "dataplane"), and rename the property to `disableForTypeSpecDataPlane`.